toptal/picasso
# [FX-4582] Migrate Modal to TailwindCSS

[FX-4582]

### Description

Migrate Modal and its components to TailwindCSS.

### How to test


- [Temploy](https://picasso.toptal.net/fx-4582-modal)
- open [story](https://picasso.toptal.net/?path=/story/overlays-modal--modal) on production and compare with [temploy](https://picasso.toptal.net/fx-4582-modal?path=/story/overlays-modal--modal). It should look the same.
- you can check SP https://github.com/toptal/staff-portal/pull/12331
- you can check CP https://github.com/toptal/client-portal/pull/8387

### Screenshots

SP
![image](https://github.com/toptal/picasso/assets/6830426/95e67bc3-1513-44b5-ae6d-e4f5772112c7)
-----
CP
![image](https://github.com/toptal/picasso/assets/6830426/231c56eb-e590-4a0a-aa4b-2dafcf0b2ec4)



### Development checks

- Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- Annotate all `props` in component with documentation
- Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)

**Breaking change**

- codemod is created and showcased in the changeset

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>


[FX-4582]: https://toptal-core.atlassian.net/browse/FX-4582?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
